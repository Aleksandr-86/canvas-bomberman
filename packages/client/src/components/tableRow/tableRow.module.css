.tableRow {
  font-weight: 400;
  font-size: 30px;
  font-family: 'Rubik Bubbles', fantasy;
  line-height: 40px;
  text-align: center;
}

.tableItem:hover {
  border: 1px solid rgb(255 255 255 / 30%);
  border-radius: 16px;
  box-shadow: 0 4px 30px hsl(120deg 50% 0% / 50%);
  cursor: pointer;
  transform: scale(1.02);
}

.leftAlignment {
  text-align: left;
}
