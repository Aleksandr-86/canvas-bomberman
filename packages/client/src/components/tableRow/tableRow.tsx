import classes from './tableRow.module.css'
import { useNavigate } from 'react-router-dom'
import { useAppDispatch } from '../../store/hooks'
import { sendDislike, sendLike } from '../../store/forumSlice'

export interface Post {
  id: number
  author: string
  body: string
  date: string
  likes?: number
  dislikes?: number
  redirect?: boolean | null
}

export const TableRow: React.FC<Post> = (props: Post) => {
  const navigate = useNavigate()
  const handleRedirect = (id: number) => {
    navigate(`/forum/${id}`)
  }
  const dispatch = useAppDispatch()

  const incrementLike = (id: number) => {
    dispatch(sendLike(id))
  }

  const incrementDislike = (id: number) => {
    dispatch(sendDislike(id))
  }

  return (
    <tr className={classes.tableItem} onClick={props.redirect? () => handleRedirect(props.id): undefined}>
      <td className={classes.tableRow}>{props.author}</td>
      <td className={classes.tableRow}>{props.body}</td>
      <td className={classes.tableRow}>{props.date}</td>
      {props.likes && <td onClick={() => incrementLike(props.id)} className={classes.tableRow}>{props.likes}</td>}
      {props.dislikes && <td onClick={() => incrementDislike(props.id)} className={classes.tableRow}>{props.dislikes}</td>}
    </tr>
  )
}
