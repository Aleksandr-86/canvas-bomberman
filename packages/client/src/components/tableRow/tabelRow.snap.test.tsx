import * as renderer from 'react-test-renderer'
import { TableRow } from './tableRow'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([])
const store = mockStore({})

const mockProps = {
  id: 1,
  author: `Kan88`,
  body: `Как пройти первый уровень`,
  date: `8.01.22'.`,
}

describe(`<TableRow/> snapshot test cases`, () => {
  test(`<TableRow/> should render <tr> parent with <td className=tableRow> childs (with mockProps)`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter initialEntries={[`/forum`]}>
            <TableRow {...mockProps} />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
