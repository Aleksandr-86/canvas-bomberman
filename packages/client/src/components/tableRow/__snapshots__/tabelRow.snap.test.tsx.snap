// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow/> snapshot test cases <TableRow/> should render <tr> parent with <td className=tableRow> childs (with mockProps) 1`] = `
<tr
  className="tableItem"
>
  <td
    className="tableRow"
  >
    Kan88
  </td>
  <td
    className="tableRow"
  >
    Как пройти первый уровень
  </td>
  <td
    className="tableRow"
  >
    8.01.22'.
  </td>
</tr>
`;
